exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/RegexpUtil.java

```diff
@@ -96,20 +145,46 @@ public static String sqlToRegexLike(
             || (nextChar == '%')
             || (nextChar == escapeChar)) {
           javaPattern.append(nextChar);
+          simplePattern.append(nextChar);
           i++;
         } else {
           throw invalidEscapeSequence(sqlPattern, i);
         }
       } else if (c == '_') {
+        // if we find _, it is not simple pattern, we are looking for only %
+        notSimple = true;
         javaPattern.append('.');
       } else if (c == '%') {
+        if (i == 0) {
+          // % at the start could potentially be one of the simple cases i.e. ENDS_WITH.
+          endsWith = true;
+        } else if (i == (len-1)) {
```

Comment:
A bit of a funky way to do this. Might was well actually wait to the end. This is why we need states (as an enum or via recursive descent.) At end:

If all constants: CONST
If one wildcard: one of the simple cases
Otherwise: COMPLEX